numpy/lib/tests/test_financial.py

```diff
@@ -344,3 +344,20 @@ def test_broadcast_decimal(self):
                                     Decimal('0'), [Decimal('0'), Decimal('0'), Decimal('1'), 'end', 'begin']),
                             [Decimal('-74.998201'), Decimal('-75.62318601'), Decimal('-75.62318601'),
                              Decimal('-76.88882405'), Decimal('-76.88882405')], 4)
+        
+    def test_rate_nan(self):
+        # Test the inputs whose outcome is nan
```

Comment:
Can you add a quick explanation to each of these tests for _why_ the outcome is nan?

Also, it might be neat to do
```
@pytest.mark.parametrize('number_type', [Decimal, float])
def test_rate_nan(self, number_type):
    # use number_type(...) instead of Decimal(...)
```

Reply:
Thanks for the review @eric-wieser . I have added explanation & parametrized the test function.